### ISSUE TITLE
Forbidden URL

### ISSUE THREAD

**Issue by nzrunner** [cyberjunky/home-assistant-garmin_connect#26]
I have managed to get the Garmin Connect integration to connect and login. But as soon as the integration connects, I get the following two lines appear in my HA Logs: -

`2022-01-04 17:15:05 ERROR (MainThread) [custom_components.garmin_connect] Connection error occurred during Garmin Connect login request: Forbidden url
2022-01-04 17:15:05 WARNING (MainThread) [homeassistant.config_entries] Config entry 'mark@domainname.com.au' for garmin_connect integration not ready yet: Forbidden url; Retrying in background`

It is possible that my PiHole has blocked something, but the HA logs don't tell me what URL is forbidden.

Any thoughts on what URL is forbidden?

**Comment by cyberjunky** [comment]
@nzrunner Something I want to know as well, it seems HA is maybe requesting an URL before login process is finished, this happens only at restart and resolves itself, I have added more debugging locally to see what exactly is happening.

**Comment by homonto** [comment]
same here:
![Screenshot 2022-01-04 at 12 23 13](https://user-images.githubusercontent.com/46562447/148058606-8e3c644d-ab93-4bc0-85a2-25489d12c711.jpg)

**Comment by cyberjunky** [comment]
The forbidden url is 'post url: https://sso.garmin.com/sso/signin'
Looking into it but as mentioned it simply retries and fetches your data.

**Comment by heinekmadsen** [comment]
Maybe this could be flagged as a warning and the if fails after the retry then an error?

**Comment by cyberjunky** [comment]
Yes looking into that, but we must retry only a fix amount, had some loops already while testing.

**Comment by jankete** [comment]
Are there any news on this? I tried to setup yesterday and after 5 retries, I end up with this:

![image](https://user-images.githubusercontent.com/22338766/168804763-8733a9c3-3343-4b36-8c0f-1cded30008f7.png)

![image](https://user-images.githubusercontent.com/22338766/168804704-2c098680-f76f-460d-85d6-e93b8891e7ee.png)

**Comment by kksligh** [comment]
This integration continually goes in and out of this state. Is there a fix for this or is this just how it currently operates?

**Comment by cyberjunky** [comment]
Closing since stale, try with latest version and open request if still an issue.
